kpi/models/import_export_task.py

```diff
@@ -689,15 +685,79 @@ def _run_task(self, messages):
             elif export_type == 'spss_labels':
                 export.to_spss_labels(output_file)
 
-        # Restore the FileField to its typical state
-        self.result.open('rb')
-        self.save(update_fields=['last_submission_time'])
+        self.result = absolute_filename
+
+        if not self.pk:
+            # In tests, exports are not saved into the DB before calling this
+            # method, thus we cannot update only specific fields.
+            self.save()
+        else:
+            self.save(update_fields=['result', 'last_submission_time'])
 
     def delete(self, *args, **kwargs):
         # removing exported file from storage
         self.result.delete(save=False)
         super().delete(*args, **kwargs)
 
+    def get_absolute_filename(self, filename: str) -> str:
+        """
+        Get absolute filename related to storage root.
+        """
+
+        storage_class = self.result.storage
+        filename = self.result.field.generate_filename(self, filename)
+
+        # We used to call `self.result.save(filename, ContentFile(b''))` before
+        # reopening the file in write mode.
+        # ```
+        #    self.result.save(filename, ContentFile(b''))
+        #    # FileField files are opened read-only by default and must be
+        #    # closed and reopened to allow writing
+        #    # https://code.djangoproject.com/ticket/13809
+        #    self.result.close()
+        #    self.result.file.close()
+        # ```
+        # But it does not work with AzureStorage.
+        # AzureStorage raises an error if we try to write into an existing file
+        # when setting `AZURE_OVERWRITES_FILES` is False.
+        # Unfortunately, `self.result.save()` does few things that we need to
+        # reimplement here:
+        # - Create parent folders (if they do not exist) for local storage
+        # - Get a unique filename if filename already exists on storage
+
+        # Copied from `FileSystemStorage._save()`
```

Comment:
:cry: guess this is what you meant by "Not super proud". If you had more time to give, this would be where to spend it. But we need a quick solution obviously.

Reply:
Indeed :-(